File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/data/remote/api/ApiCepApi.java
package com.example.challenge_consult_cep.feature.search_cep.data.remote.api;

import com.example.challenge_consult_cep.feature.search_cep.data.remote.dto.ApiCepDTO;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiCepApi {

    @GET("https://cdn.apicep.com/file/apicep/{cep}.json")
    Single<ApiCepDTO> getCep(@Path("cep") String cep);
}

File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/data/remote/api/ApiCepApi.kt
package com.example.challenge_consult_cep.feature.search_cep.data.remote.api

import com.example.challenge_consult_cep.feature.search_cep.data.remote.dto.ApiCepDTO
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiCepApi {

    @GET("https://cdn.apicep.com/file/apicep/{cep}.json")
    suspend fun getCep(
        @Path("cep") cep:String
    ): ApiCepDTO
}